test/src/util/epoch_number_with_fraction.rs

```diff
@@ -0,0 +1,16 @@
+use ckb_jsonrpc_types::EpochNumberWithFraction;
+
+// The highest 16 bits represent the epoch length
+// The next 16 bits represent the current block index in the epoch, starting from 0.
+// The lowest 24 bits represent the current epoch number.
+pub fn get_epoch_number(epoch: EpochNumberWithFraction) -> u64 {
+    epoch.value() & 0xFFFFFF
+}
+
+pub fn get_epoch_index(epoch: EpochNumberWithFraction) -> u64 {
+    (epoch.value() >> 24) & 0xFFFF
+}
+
+pub fn get_epoch_length(epoch: EpochNumberWithFraction) -> u64 {
+    (epoch.value() >> 40) & 0xFFFF
+}
```

Comment:
Just a suggestion: add the corresponding implementation to `ckb-jsonrpc-types::EpochNumberWithFraction`, of cause, in another PR.

Reply:
Thanks for your suggestion, create a new PR for this: https://github.com/nervosnetwork/ckb/pull/2495

Reply:
Please rebase when #2495 is merged